### PR TITLE
Update EclipseLink 2.7.11 -> 2.7.12

### PR DESCRIPTION
**Pull request by tkburroughs**
Updating EclipseLink from 2.7.11 to 2.7.12.

for #22796
for #23678
for #23942
for #23950

### CODE REVIEW

**Comment by tkburroughs** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jpa.tests.eclipselink_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.entity_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.query_jpa_2.2_fat

**Comment by tkburroughs** [comment]
#build #spawn.fullfat.buckets=com.ibm.ws.jpa.tests.eclipselink_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.entity_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.query_jpa_2.2_fat

**Comment by tkburroughs** [comment]
#libby

**Review by brideck — APPROVED** [review]

**Comment by posmith** [comment]
L2 message review completed

**Review by jgrassel — APPROVED** [review]

**Comment by tkburroughs** [comment]
#libby
#build #spawn.fullfat.buckets=com.ibm.ws.jpa.tests.eclipselink_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.entity_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.query_jpa_2.2_fat

**Comment by tkburroughs** [comment]
#libby
#build #spawn.fullfat.buckets=com.ibm.ws.jpa.tests.eclipselink_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.entity_jpa_2.2_fat,com.ibm.ws.jpa.tests.spec10.query_jpa_2.2_fat

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/BeanValidationExceptionResource.nlsprops [hunk 1]
```diff
@@ -31,10 +31,10 @@
 # -------------------------------------------------------------------------------------------------
```
Comment (jim-krueger) [inline dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/BeanValidationExceptionResource.nlsprops:31]: Add 2023 to copyright

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops [hunk 2]
```diff
@@ -39,13 +39,13 @@
```
Comment (jim-krueger) [inline dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops:39]: Add 2023 to copyright

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/DBWSExceptionResource.nlsprops [hunk 3]
```diff
@@ -29,29 +29,29 @@
 #   (single quote must be coded as one single quote '). 
 #
 # -------------------------------------------------------------------------------------------------
```
Comment (jim-krueger) [inline dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/DBWSExceptionResource.nlsprops:31]: Add 2023 to copyright

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops (hunk 2 above)
Comment (jim-krueger) [inline dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops:39]: Add 2023 to copyright

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/internal/localization/i18n/TraceLocalizationResource.nlsprops [hunk 4]
```diff
@@ -0,0 +1,1024 @@
+###############################################################################
+# Copyright (c) 2014 IBM Corporation and others.
```
Comment (jim-krueger) [inline dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/internal/localization/i18n/TraceLocalizationResource.nlsprops:2]: Add 2023

**Review by jim-krueger — COMMENTED** [review]
Looks good, just some copyright stuff

**Review by jim-krueger — APPROVED** [review]
The copyrights I'd pointed out could not be addressed.   Approving.

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/BeanValidationExceptionResource.nlsprops (hunk 1 above)
Comment (tkburroughs) [reply dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/BeanValidationExceptionResource.nlsprops:31]: A nlsprops verification tool in the build does not currently allow the copyright to be updated.

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops (hunk 2 above)
Comment (tkburroughs) [reply dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops:39]: A nlsprops verification tool in the build does not currently allow the copyright to be updated.

Comment (tkburroughs) [reply dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/exceptions/i18n/ConcurrencyExceptionResource.nlsprops:39]: A nlsprops verification tool in the build does not currently allow the copyright to be updated.

File: dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/internal/localization/i18n/TraceLocalizationResource.nlsprops (hunk 4 above)
Comment (tkburroughs) [reply dev/com.ibm.websphere.appserver.thirdparty.eclipselink.2.7/resources/org/eclipse/persistence/internal/localization/i18n/TraceLocalizationResource.nlsprops:2]: A nlsprops verification tool in the build does not currently allow the copyright to be updated.

**Pull request merged**